src/components/text-input/TextInput.tsx

```diff
@@ -15,31 +15,27 @@ const TextInput: React.FC<PTextInput> = ({
     leadingIcon,
     trailingIcon,
     width,
+    onClear,
     ...rest
 }: PTextInput) => {
     const { iconSize, labelSize } = TEXT_INPUT_VALUES_MAPPING[size];
 
     const hasLabel = Utils.isString(label) && label.length > 0;
     const hasPlaceholder = Utils.isString(placeholder) && placeholder.length > 0;
-
-    const [value, setValue] = useState();
+    const isClearable = Utils.isFunction(onClear);
+    const onClearInput = (): void => (isClearable ? onClear : null);
 
     return (
-        <TextInputRoot size={size} width={width} {...rest}>
+        <TextInputRoot size={size} width={width} leadingIcon={leadingIcon} {...rest}>
             {leadingIcon !== 'none' && <Icon glyph={leadingIcon} size={iconSize} />}
-            <input
-                className={'input__field'}
-                value={value}
-                onChange={(event): void => setValue(event.target.value)}
-                placeholder={hasPlaceholder ? placeholder : ''}
-            />
+            <input className={'input__field'} placeholder={hasPlaceholder ? placeholder : ''} />
             {hasLabel && (
                 <Text element={'span'} size={labelSize} className={'input__label'}>
                     {label}
                 </Text>
             )}
             {trailingIcon !== 'none' && (
-                <Icon glyph={trailingIcon} size={iconSize} onClick={(): void => setValue('')} />
+                <Icon glyph={trailingIcon} size={iconSize} onClick={onClearInput} />
```

Comment:
the `trailingIcon !== 'none'` only works with mandatory properties.
change it to `trailingIcon && trailingIcon !== 'none'`